test/_ui-integration/nrm/validations.spec.js

```diff
@@ -609,6 +609,40 @@ describe('validation checks of the nrm journey', () => {
       expect(validationSummary.html())
         .to.match(/You must select an option/);
     });
+
+    it('does not pass the modern slavery indicators page if more than 15000 entered', async () => {
+      const URI = '/any-other-pvs';
+      await initSession(URI);
+      await passStep(URI, {
+        'any-other-pvs': 'yes',
+        'other-potential-victims-yes-details': 'a'.repeat(15001)
+      });
+
+      const res = await getUrl(URI);
+      const docu = await parseHtml(res);
+      const validationSummary = docu.find('.govuk-error-summary');
+
+      expect(validationSummary.length === 1).to.be.true;
+      expect(validationSummary.html())
+        .to.match(/Details of other potential victims must be 15000 characters or less/);
+    });
+
+    it('does not pass the modern slavery indicators page if more than 15000 entered', async () => {
+      const URI = '/any-other-pvs';
+      await initSession(URI);
+      await passStep(URI, {
+        'any-other-pvs': 'not-sure',
+        'other-potential-victims-not-sure-details': 'a'.repeat(15001)
+      });
+
+      const res = await getUrl(URI);
+      const docu = await parseHtml(res);
+      const validationSummary = docu.find('.govuk-error-summary');
+
+      expect(validationSummary.length === 1).to.be.true;
+      expect(validationSummary.html())
+        .to.match(/Details of other potential victims must be 15000 characters or less/);
+    });
```

Comment:
not sure this work but you get the idea 🙂 
```suggestion
    const testFieldValidation = async (field, value, errorMessage) => {
  const URI = '/any-other-pvs';
  await initSession(URI);
  const stepData = { 'any-other-pvs': field.includes('yes') ? 'yes' : 'not-sure', [field]: value };
  await passStep(URI, stepData);

  const res = await getUrl(URI);
  const docu = await parseHtml(res);
  const validationSummary = docu.find('.govuk-error-summary');

  expect(validationSummary.length === 1).to.be.true;
  expect(validationSummary.html()).to.match(new RegExp(errorMessage));
};

it('does not pass the modern slavery indicators page if more than 15000 entered', async () => {
  await testFieldValidation('other-potential-victims-yes-details', 'a'.repeat(15001), 'Details of other potential victims must be 15000 characters or less');
});

it('does not pass the modern slavery indicators page if more than 15000 entered', async () => {
  await testFieldValidation('other-potential-victims-not-sure-details', 'a'.repeat(15001), 'Details of other potential victims must be 15000 characters or less');
});
```

Reply:
Thanks Sulthan, I have amended this as suggested